src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ReferralDetails.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity
+
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.OneToOne
+
+@Entity
+data class ReferralDetails (
+  @Id val id: UUID,
+  @OneToOne var supersededBy: ReferralDetails?,
```

Comment:
_performance question_
Would loading this field cause recursive resolution of the entire chain? As in, the first detail would load the next, then the next, etc...? Asking as we don't need that behaviour in typical use (we might be able to achieve updates to this by a specific repository update function)

Reply:
dealt with in cd25dfe3a6467f6f42d7b5402c2189f9fca3e18d